### ISSUE TITLE
It's not possible to make a connection with Shadowsocks from outside the LAN

### ISSUE THREAD

**Issue by Frepke** [qdm12/gluetun#82]
Just a reminder :)

**Comment by qdm12** [comment]
Do you think it would be better:
1. Have a `SHADOWSOCKS_SUBNETS` environment variable to define a comma separated list of IP address ranges (i.e. 10.0.0.0/24 or 0.0.0.0/0) for the firewall to allow incoming traffic to the Shadowsocks 8388 TCP+UDP port
2. Have a `SHADOWSOCKS_EXTERNAL` environment variable set to on/off to allow incoming traffic from `0.0.0.0/0` (anywhere) on the Shadowsocks port

Thanks :)

Also have you tried setting `EXTRA_SUBNETS=0.0.0.0/0` to access Shadowsocks from outside? It's not recommended as it disables some of firewall setup, but you could have a test.

**Comment by Frepke** [comment]
Hello Quentin,

With EXTRA_SUBNETS=0.0.0.0/0 I can't connect to shadowsocks anymore.
I think SHADOWSOCKS_EXTERNAL environment variable set to on/off is the easiest option, it's trough LAN or WAN, there aren't more options for me.


kr.,
Frepke

**Comment by qdm12** [comment]
In the end, I just added an input rule allowing any traffic to the Shadowsocks port (or tinyproxy port) if Shadowsocks (or tinyproxy) is enabled. It should not cause any security issue anyway.

However, I did not have time to test it, feel free to try it 👍  And re-open if it's not working, maybe the firewall rules are not well ordered, it's really a quick hack I did 😄

**Comment by Frepke** [comment]
Unfortunately it's not working for me, even when I expose my host completely to the internet.

**Comment by qdm12** [comment]
Ok, thanks for trying. I'll test it/fix it this evening.

**Comment by Frepke** [comment]
I couldn’t re-open thus issue myselve 😔

**Comment by qdm12** [comment]
Just a little update, I need this fixed for myself as well ASAP so I will definitely fix that up today/tomorrow. I'll close the issue once done.

**Comment by qdm12** [comment]
Hi @Frepke I tried quite hard the last few days with Shadowsocks without success, but the firewall does not seem to block Shadowsocks. This is really bugging me.

I tried basically to run the container and then run `docker exec -it <container_name> watch -n 1 -d iptables -nvL` to see live iptables changes in another terminal.

Anyway, do you mind trying with Tinyproxy (I don't have that setup on my phone etc. and running out of time 😄 )? I really need this as well haha so I better figure something out too.

**Comment by qdm12** [comment]
Ok I found the problem comes from the network routes.

For now, it's only adding what is specified in `EXTRA_SUBNETS` but setting `0.0.0.0/0` wont' work, we need to set a default route somehow. I'll ask the Reddit iptables gurus, maybe they have an idea how to achieve this.

**Comment by Phorze** [comment]
Not sure if i am encountering a similar issue:

```
shadowsocks:  2020-03-26 21:40:32 INFO: initializing ciphers... chacha20-ietf-poly1305
shadowsocks:  2020-03-26 21:40:32 INFO: using nameserver: 127.0.0.1
shadowsocks:  2020-03-26 21:40:32 INFO: tcp server listening at 0.0.0.0:8388
shadowsocks:  2020-03-26 21:40:32 INFO: udp server listening at 0.0.0.0:8388
....
shadowsocks:  2020-03-26 21:41:05 INFO: [8388] connect to www.youtube.com:443
shadowsocks:  2020-03-26 21:41:05 INFO: failed to lookup v6 address Could not contact DNS servers
shadowsocks:  2020-03-26 21:41:05 INFO: failed to lookup v4 address Could not contact DNS servers
```

using:
https://github.com/EasyPi/docker-shadowsocks-libev 
as client from the same docker host (udp enabled)

**Comment by qdm12** [comment]
No the issue is about accessing the container from any IP address, and not just from the network(s) specified by `EXTRA_SUBNETS`. Can you open another issue? :+1:  And maybe you're not tunneling udp? Some Shadowsocks client don't tunnel udp.

**Comment by qdm12** [comment]
Will add when [this Reddit post](https://www.reddit.com/r/HomeNetworking/comments/fzeqxt/which_route_for_any_ip_to_connect_to_an_openvpn/) is (hopefully) answered 😉

**Comment by Frepke** [comment]
> 
> 
> Will add when [this Reddit post](https://www.reddit.com/r/HomeNetworking/comments/fzeqxt/which_route_for_any_ip_to_connect_to_an_openvpn/) is (hopefully) answered 😉

Any news on this one?

Kr.,
Frepke

**Comment by qdm12** [comment]
I'll try setting *all* the public IP address CIDR ranges to the EXTRA_SUBNETS (https://serverfault.com/questions/304781/cidr-ranges-for-everything-except-rfc1918). I'll set a variable so you don't have to go through adding them all :wink: You can test it relatively easily if you want.

**Comment by Frepke** [comment]
I tried my ISP ip as extra subnet, but that didn't work

**Comment by qdm12** [comment]
Try with

```
EXTRA_SUBNETS=0.0.0.0/5,8.0.0.0/7,11.0.0.0/8,12.0.0.0/6,16.0.0.0/4,32.0.0.0/3,64.0.0.0/2,128.0.0.0/3,160.0.0.0/5,168.0.0.0/6,172.0.0.0/12,172.32.0.0/11,172.64.0.0/10,172.128.0.0/9,173.0.0.0/8,174.0.0.0/7,176.0.0.0/4,192.0.0.0/9, 192.128.0.0/11,192.160.0.0/13,192.169.0.0/16,192.170.0.0/15,192.172.0.0/14,192.176.0.0/12,192.192.0.0/10,193.0.0.0/8,194.0.0.0/7,196.0.0.0/6,200.0.0.0/5,208.0.0.0/4
```

**Comment by Frepke** [comment]
> Try with
> 
> ```
> EXTRA_SUBNETS=0.0.0.0/5,8.0.0.0/7,11.0.0.0/8,12.0.0.0/6,16.0.0.0/4,32.0.0.0/3,64.0.0.0/2,128.0.0.0/3,160.0.0.0/5,168.0.0.0/6,172.0.0.0/12,172.32.0.0/11,172.64.0.0/10,172.128.0.0/9,173.0.0.0/8,174.0.0.0/7,176.0.0.0/4,192.0.0.0/9, 192.128.0.0/11,192.160.0.0/13,192.169.0.0/16,192.170.0.0/15,192.172.0.0/14,192.176.0.0/12,192.192.0.0/10,193.0.0.0/8,194.0.0.0/7,196.0.0.0/6,200.0.0.0/5,208.0.0.0/4
> ```
> 
> That should allow any non private IP address yo in connect. Although I'm not 100% sure of the implications openvpn wise / security wise but I guess it should be fine.

```
ERROR	could not parse subnet " 192.128.0.0/11" from environment variable with key EXTRA_SUBNETS: invalid CIDR address:  192.128.0.0/11

**Comment by Frepke** [comment]
> Try with
> 
> ```
> EXTRA_SUBNETS=0.0.0.0/5,8.0.0.0/7,11.0.0.0/8,12.0.0.0/6,16.0.0.0/4,32.0.0.0/3,64.0.0.0/2,128.0.0.0/3,160.0.0.0/5,168.0.0.0/6,172.0.0.0/12,172.32.0.0/11,172.64.0.0/10,172.128.0.0/9,173.0.0.0/8,174.0.0.0/7,176.0.0.0/4,192.0.0.0/9, 192.128.0.0/11,192.160.0.0/13,192.169.0.0/16,192.170.0.0/15,192.172.0.0/14,192.176.0.0/12,192.192.0.0/10,193.0.0.0/8,194.0.0.0/7,196.0.0.0/6,200.0.0.0/5,208.0.0.0/4
> ```
> 
> That should allow any non private IP address yo in connect. Although I'm not 100% sure of the implications openvpn wise / security wise but I guess it should be fine.

1st, in front of 192.128.x.x is a space which ends in an error.

2nd, I now can connect to my server from outside my lan but the vpn isn't working anymore. From outside, local and connected containers I only get my ISP ip

```
2020-05-20T14:27:06.992+0200	INFO	Launching standard output merger
2020-05-20T14:27:06.993+0200	INFO	routing: adding 0.0.0.0/5 as route via eth0
2020-05-20T14:27:07.001+0200	INFO	routing: adding 8.0.0.0/7 as route via eth0
2020-05-20T14:27:07.002+0200	INFO	routing: adding 11.0.0.0/8 as route via eth0
2020-05-20T14:27:07.004+0200	INFO	routing: adding 12.0.0.0/6 as route via eth0
2020-05-20T14:27:07.009+0200	INFO	routing: adding 16.0.0.0/4 as route via eth0
2020-05-20T14:27:07.014+0200	INFO	routing: adding 32.0.0.0/3 as route via eth0
2020-05-20T14:27:07.017+0200	INFO	routing: adding 64.0.0.0/2 as route via eth0
2020-05-20T14:27:07.021+0200	INFO	routing: adding 128.0.0.0/3 as route via eth0
2020-05-20T14:27:07.026+0200	INFO	routing: adding 160.0.0.0/5 as route via eth0
2020-05-20T14:27:07.029+0200	INFO	routing: adding 168.0.0.0/6 as route via eth0
2020-05-20T14:27:07.031+0200	INFO	routing: adding 172.0.0.0/12 as route via eth0
2020-05-20T14:27:07.034+0200	INFO	routing: adding 172.32.0.0/11 as route via eth0
2020-05-20T14:27:07.036+0200	INFO	routing: adding 172.64.0.0/10 as route via eth0
2020-05-20T14:27:07.039+0200	INFO	routing: adding 172.128.0.0/9 as route via eth0
2020-05-20T14:27:07.041+0200	INFO	routing: adding 173.0.0.0/8 as route via eth0
2020-05-20T14:27:07.044+0200	INFO	routing: adding 174.0.0.0/7 as route via eth0
2020-05-20T14:27:07.050+0200	INFO	routing: adding 176.0.0.0/4 as route via eth0
2020-05-20T14:27:07.055+0200	INFO	routing: adding 192.0.0.0/9 as route via eth0
2020-05-20T14:27:07.059+0200	INFO	routing: adding 192.128.0.0/11 as route via eth0
2020-05-20T14:27:07.065+0200	INFO	routing: adding 192.160.0.0/13 as route via eth0
2020-05-20T14:27:07.069+0200	INFO	routing: adding 192.169.0.0/16 as route via eth0
2020-05-20T14:27:07.072+0200	INFO	routing: adding 192.170.0.0/15 as route via eth0
2020-05-20T14:27:07.078+0200	INFO	routing: adding 192.172.0.0/14 as route via eth0
2020-05-20T14:27:07.081+0200	INFO	routing: adding 192.176.0.0/12 as route via eth0
2020-05-20T14:27:07.084+0200	INFO	routing: adding 192.192.0.0/10 as route via eth0
2020-05-20T14:27:07.088+0200	INFO	routing: adding 193.0.0.0/8 as route via eth0
2020-05-20T14:27:07.094+0200	INFO	routing: adding 194.0.0.0/7 as route via eth0
2020-05-20T14:27:07.097+0200	INFO	routing: adding 196.0.0.0/6 as route via eth0
2020-05-20T14:27:07.101+0200	INFO	routing: adding 200.0.0.0/5 as route via eth0
2020-05-20T14:27:07.106+0200	INFO	routing: adding 208.0.0.0/4 as route via eth0
2020-05-20T14:27:07.110+0200	INFO	routing: adding 10.54.1.0/24 as route via eth0
2020-05-20T14:27:07.118+0200	INFO	firewall configurator: clearing all rules
2020-05-20T14:27:07.170+0200	INFO	firewall configurator: blocking all traffic
2020-05-20T14:27:07.194+0200	INFO	firewall configurator: creating general rules
2020-05-20T14:27:07.228+0200	INFO	firewall configurator: allowing output traffic to VPN server 46.166.137.218 through eth0 on port udp 1197
2020-05-20T14:27:07.239+0200	INFO	firewall configurator: allowing output traffic to VPN server 46.166.138.138 through eth0 on port udp 1197
2020-05-20T14:27:07.249+0200	INFO	firewall configurator: allowing output traffic to VPN server 46.166.138.161 through eth0 on port udp 1197
2020-05-20T14:27:07.254+0200	INFO	firewall configurator: allowing output traffic to VPN server 46.166.138.172 through eth0 on port udp 1197
2020-05-20T14:27:07.258+0200	INFO	firewall configurator: allowing output traffic to VPN server 46.166.188.211 through eth0 on port udp 1197
2020-05-20T14:27:07.262+0200	INFO	firewall configurator: allowing output traffic to VPN server 46.166.188.218 through eth0 on port udp 1197
2020-05-20T14:27:07.265+0200	INFO	firewall configurator: allowing output traffic to VPN server 46.166.190.178 through eth0 on port udp 1197
2020-05-20T14:27:07.272+0200	INFO	firewall configurator: allowing output traffic to VPN server 46.166.190.223 through eth0 on port udp 1197
2020-05-20T14:27:07.277+0200	INFO	firewall configurator: allowing output traffic to VPN server 46.166.190.227 through eth0 on port udp 1197
2020-05-20T14:27:07.281+0200	INFO	firewall configurator: allowing output traffic to VPN server 46.166.190.230 through eth0 on port udp 1197
2020-05-20T14:27:07.300+0200	INFO	firewall configurator: accepting input and output traffic for 172.17.0.0/16
2020-05-20T14:27:07.314+0200	INFO	firewall configurator: accepting input traffic through eth0 from 0.0.0.0/5 to 172.17.0.0/16
2020-05-20T14:27:07.319+0200	INFO	firewall configurator: accepting output traffic through eth0 from 172.17.0.0/16 to 0.0.0.0/5
2020-05-20T14:27:07.323+0200	INFO	firewall configurator: accepting input traffic through eth0 from 8.0.0.0/7 to 172.17.0.0/16
2020-05-20T14:27:07.328+0200	INFO	firewall configurator: accepting output traffic through eth0 from 172.17.0.0/16 to 8.0.0.0/7
2020-05-20T14:27:07.331+0200	INFO	firewall configurator: accepting input traffic through eth0 from 11.0.0.0/8 to 172.17.0.0/16
2020-05-20T14:27:07.335+0200	INFO	firewall configurator: accepting output traffic through eth0 from 172.17.0.0/16 to 11.0.0.0/8
2020-05-20T14:27:07.343+0200	INFO	firewall configurator: accepting input traffic through eth0 from 12.0.0.0/6 to 172.17.0.0/16
2020-05-20T14:27:07.348+0200	INFO	firewall configurator: accepting output traffic through eth0 from 172.17.0.0/16 to 12.0.0.0/6
2020-05-20T14:27:07.352+0200	INFO	firewall configurator: accepting input traffic through eth0 from 16.0.0.0/4 to 172.17.0.0/16
2020-05-20T14:27:07.355+0200	INFO	firewall configurator: accepting output traffic through eth0 from 172.17.0.0/16 to 16.0.0.0/4
2020-05-20T14:27:07.360+0200	INFO	firewall configurator: accepting input traffic through eth0 from 32.0.0.0/3 to 172.17.0.0/16
2020-05-20T14:27:07.364+0200	INFO	firewall configurator: accepting output traffic through eth0 from 172.17.0.0/16 to 32.0.0.0/3
2020-05-20T14:27:07.372+0200	INFO	firewall configurator: accepting input traffic through eth0 from 64.0.0.0/2 to 172.17.0.0/16
2020-05-20T14:27:07.378+0200	INFO	firewall configurator: accepting output traffic through eth0 from 172.17.0.0/16 to 64.0.0.0/2
2020-05-20T14:27:07.382+0200	INFO	firewall configurator: accepting input traffic through eth0 from 128.0.0.0/3 to 172.17.0.0/16
2020-05-20T14:27:07.386+0200	INFO	firewall configurator: accepting output traffic through eth0 from 172.17.0.0/16 to 128.0.0.0/3
2020-05-20T14:27:07.391+0200	INFO	firewall configurator: accepting input traffic through eth0 from 160.0.0.0/5 to 172.17.0.0/16
2020-05-20T14:27:07.396+0200	INFO	firewall configurator: accepting output traffic through eth0 from 172.17.0.0/16 to 160.0.0.0/5
2020-05-20T14:27:07.403+0200	INFO	firewall configurator: accepting input traffic through eth0 from 168.0.0.0/6 to 172.17.0.0/16
2020-05-20T14:27:07.410+0200	INFO	firewall configurator: accepting output traffic through eth0 from 172.17.0.0/16 to 168.0.0.0/6
2020-05-20T14:27:07.416+0200	INFO	firewall configurator: accepting input traffic through eth0 from 172.0.0.0/12 to 172.17.0.0/16
2020-05-20T14:27:07.420+0200	INFO	firewall configurator: accepting output traffic through eth0 from 172.17.0.0/16 to 172.0.0.0/12
2020-05-20T14:27:07.424+0200	INFO	firewall configurator: accepting input traffic through eth0 from 172.32.0.0/11 to 172.17.0.0/16
2020-05-20T14:27:07.427+0200	INFO	firewall configurator: accepting output traffic through eth0 from 172.17.0.0/16 to 172.32.0.0/11
2020-05-20T14:27:07.431+0200	INFO	firewall configurator: accepting input traffic through eth0 from 172.64.0.0/10 to 172.17.0.0/16
2020-05-20T14:27:07.437+0200	INFO	firewall configurator: accepting output traffic through eth0 from 172.17.0.0/16 to 172.64.0.0/10
2020-05-20T14:27:07.443+0200	INFO	firewall configurator: accepting input traffic through eth0 from 172.128.0.0/9 to 172.17.0.0/16
2020-05-20T14:27:07.454+0200	INFO	firewall configurator: accepting output traffic through eth0 from 172.17.0.0/16 to 172.128.0.0/9
2020-05-20T14:27:07.460+0200	INFO	firewall configurator: accepting input traffic through eth0 from 173.0.0.0/8 to 172.17.0.0/16
2020-05-20T14:27:07.468+0200	INFO	firewall configurator: accepting output traffic through eth0 from 172.17.0.0/16 to 173.0.0.0/8
2020-05-20T14:27:07.472+0200	INFO	firewall configurator: accepting input traffic through eth0 from 174.0.0.0/7 to 172.17.0.0/16
2020-05-20T14:27:07.476+0200	INFO	firewall configurator: accepting output traffic through eth0 from 172.17.0.0/16 to 174.0.0.0/7
2020-05-20T14:27:07.482+0200	INFO	firewall configurator: accepting input traffic through eth0 from 176.0.0.0/4 to 172.17.0.0/16
2020-05-20T14:27:07.487+0200	INFO	firewall configurator: accepting output traffic through eth0 from 172.17.0.0/16 to 176.0.0.0/4
2020-05-20T14:27:07.501+0200	INFO	firewall configurator: accepting input traffic through eth0 from 192.0.0.0/9 to 172.17.0.0/16
2020-05-20T14:27:07.517+0200	INFO	firewall configurator: accepting output traffic through eth0 from 172.17.0.0/16 to 192.0.0.0/9
2020-05-20T14:27:07.521+0200	INFO	firewall configurator: accepting input traffic through eth0 from 192.128.0.0/11 to 172.17.0.0/16
2020-05-20T14:27:07.526+0200	INFO	firewall configurator: accepting output traffic through eth0 from 172.17.0.0/16 to 192.128.0.0/11
2020-05-20T14:27:07.532+0200	INFO	firewall configurator: accepting input traffic through eth0 from 192.160.0.0/13 to 172.17.0.0/16
2020-05-20T14:27:07.537+0200	INFO	firewall configurator: accepting output traffic through eth0 from 172.17.0.0/16 to 192.160.0.0/13
2020-05-20T14:27:07.544+0200	INFO	firewall configurator: accepting input traffic through eth0 from 192.169.0.0/16 to 172.17.0.0/16
2020-05-20T14:27:07.550+0200	INFO	firewall configurator: accepting output traffic through eth0 from 172.17.0.0/16 to 192.169.0.0/16
2020-05-20T14:27:07.554+0200	INFO	firewall configurator: accepting input traffic through eth0 from 192.170.0.0/15 to 172.17.0.0/16
2020-05-20T14:27:07.561+0200	INFO	firewall configurator: accepting output traffic through eth0 from 172.17.0.0/16 to 192.170.0.0/15
2020-05-20T14:27:07.565+0200	INFO	firewall configurator: accepting input traffic through eth0 from 192.172.0.0/14 to 172.17.0.0/16
2020-05-20T14:27:07.568+0200	INFO	firewall configurator: accepting output traffic through eth0 from 172.17.0.0/16 to 192.172.0.0/14
2020-05-20T14:27:07.579+0200	INFO	firewall configurator: accepting input traffic through eth0 from 192.176.0.0/12 to 172.17.0.0/16
2020-05-20T14:27:07.583+0200	INFO	firewall configurator: accepting output traffic through eth0 from 172.17.0.0/16 to 192.176.0.0/12
2020-05-20T14:27:07.590+0200	INFO	firewall configurator: accepting input traffic through eth0 from 192.192.0.0/10 to 172.17.0.0/16
2020-05-20T14:27:07.600+0200	INFO	firewall configurator: accepting output traffic through eth0 from 172.17.0.0/16 to 192.192.0.0/10
2020-05-20T14:27:07.604+0200	INFO	firewall configurator: accepting input traffic through eth0 from 193.0.0.0/8 to 172.17.0.0/16
2020-05-20T14:27:07.610+0200	INFO	firewall configurator: accepting output traffic through eth0 from 172.17.0.0/16 to 193.0.0.0/8
2020-05-20T14:27:07.617+0200	INFO	firewall configurator: accepting input traffic through eth0 from 194.0.0.0/7 to 172.17.0.0/16
2020-05-20T14:27:07.622+0200	INFO	firewall configurator: accepting output traffic through eth0 from 172.17.0.0/16 to 194.0.0.0/7
2020-05-20T14:27:07.628+0200	INFO	firewall configurator: accepting input traffic through eth0 from 196.0.0.0/6 to 172.17.0.0/16
2020-05-20T14:27:07.632+0200	INFO	firewall configurator: accepting output traffic through eth0 from 172.17.0.0/16 to 196.0.0.0/6
2020-05-20T14:27:07.642+0200	INFO	firewall configurator: accepting input traffic through eth0 from 200.0.0.0/5 to 172.17.0.0/16
2020-05-20T14:27:07.646+0200	INFO	firewall configurator: accepting output traffic through eth0 from 172.17.0.0/16 to 200.0.0.0/5
2020-05-20T14:27:07.650+0200	INFO	firewall configurator: accepting input traffic through eth0 from 208.0.0.0/4 to 172.17.0.0/16
2020-05-20T14:27:07.653+0200	INFO	firewall configurator: accepting output traffic through eth0 from 172.17.0.0/16 to 208.0.0.0/4
2020-05-20T14:27:07.656+0200	INFO	firewall configurator: accepting input traffic through eth0 from 10.54.1.0/24 to 172.17.0.0/16

**Comment by qdm12** [comment]
Oops there is probably a space that sneaked there (on mobile now) after a `,`

**Comment by qdm12** [comment]
Ah bummer. Maybe excluding VPN server ips would solve it.

Can you try getting your (out of lan) public ip address (i. e. `1.2.3.4`) and set it as `EXTRA_SUBNETS=1.2.3.4/32`? That could be a workaround for now perhaps.

**Comment by Frepke** [comment]
I did another test.
I took my phones ip address when it was connected to my cellular network x.x.x.0/24 and set this as EXTRA_SUBNET. That worked for me.

**Comment by qdm12** [comment]
Cool. That most likely means adding routes for all public IP addresses except the vpn server ip address(es) would work. Now I need to setup something to calculate these subnets ranges dynamically :smile: I'll do that this weekend!

**Comment by lavaguy1** [comment]
Wouldn't it be easier to do this with port-forwarding on the local router/firewall in front of the gluetun box/vm? That way, nobody can  directly touch your SOCKS server when you aren't looking... Or am I misunderstanding the configuration/topology you are trying to solve for?

**Comment by qdm12** [comment]
Port forwarding doesn't work because gluetun still sees the source IP address as external and not as your router.

The problem is openvpn (rightfully) sets all to be routed through the VPN tunnel interface `tun0`. The EXTRA_SUBNETS variable specifies a subnet of ips that should be routed through the docker local network interface (usually `eth0`). Now what we want is ideally to have all traffic routed through both interfaces, but I have not really a clue on how to do this yet.

Although this issue is really something annoying me. I have to run wireguard + shadowsocks to tunnel remotely to a VPN server lol... a bit overkill.

**Comment by qdm12** [comment]
Note to future self, read https://medium.com/@yingshaoxo/mastering-iptables-ip-iproute2-b0e1aeabc186 there must be a niche ip routing way to access the container from everywhere.
All packets should be sent through the tunnel, but all packets should be received through any interfaces.

I really need this fixed, I got some fun features I'd like to add blocked by this.

**Comment by qdm12** [comment]
I trolled around for a few hours today again. No way to make it work, it really does conflict with the VPN client routing. It's one or the other.

As a workaround, I'll get working on a simple TCP proxy container (I may add UDP later) you would run in the same Docker network as Gluetun which will relay the packets to Gluetun (well, services using Gluetun's network stack especially). This works as they share the same IP gateway in the Docker virtual network. As that TCP proxy container is not using a VPN client, its IP routing is 'normal' and can thus be accessed from anywhere unlike Gluetun.

**Comment by qdm12** [comment]
Hello people, that Reddit post finally got an answer which WORKS! You will now be able to access gluetun from outside your pre-defined LAN network and there won't be a need for `EXTRA_SUBNETS`.

It might take me a few days to implement, but it's coming soon!
